test/common/upstream/load_balancer_benchmark.cc

```diff
@@ -97,13 +98,18 @@ class LeastRequestTester : public BaseTester {
   std::unique_ptr<LeastRequestLoadBalancer> lb_;
 };
 
-void benchmarkRoundRobinLoadBalancerBuild(benchmark::State& state) {
+void benchmarkRoundRobinLoadBalancerBuild(::benchmark::State& state) {
+  const uint64_t num_hosts = state.range(0);
+  const uint64_t weighted_subset_percent = state.range(1);
+  const uint64_t weight = state.range(2);
+
+  if (benchmark::skipExpensiveBenchmarks() && num_hosts > 10000) {
```

Comment:
I think we could lower the `10000` constant even further in this case, if it improves execution time. Really, it's only worth running the test with more than a few hosts in a release build.

Reply:
The round robin benchmarks are relatively fast.  It's possible that we're missing tools that allow us to just run 1 set of parameters per benchmark in test mode so we at least know that each of the benchmarks are not broken by recent changes.  I'm fine reducing the threshold for the RR test further, but the effects on total runtime should be relatively small.  If we want to reduce runtime further I recommend shaving off a few benchmarks in other categories, including the three benchmarkRingHashLoadBalancerBuildRing/*/256000 which would save about 5 seconds.

benchmarkRoundRobinLoadBalancerBuild/1/0/1                        9.57 ms
benchmarkRoundRobinLoadBalancerBuild/500/0/1                      11.7 ms
benchmarkRoundRobinLoadBalancerBuild/500/50/50                    20.1 ms
benchmarkRoundRobinLoadBalancerBuild/500/100/50                   11.8 ms
benchmarkRoundRobinLoadBalancerBuild/2500/0/1                     19.3 ms
benchmarkRoundRobinLoadBalancerBuild/2500/50/50                   80.4 ms
benchmarkRoundRobinLoadBalancerBuild/2500/100/50                  19.4 ms
benchmarkRoundRobinLoadBalancerBuild/10000/0/1                    57.7 ms
benchmarkRoundRobinLoadBalancerBuild/10000/50/50                   339 ms
benchmarkRoundRobinLoadBalancerBuild/10000/100/50                 60.7 ms
--- cut off
benchmarkRoundRobinLoadBalancerBuild/25000/0/1                     137 ms
benchmarkRoundRobinLoadBalancerBuild/25000/50/50                   506 ms
benchmarkRoundRobinLoadBalancerBuild/25000/100/50                  137 ms
benchmarkRoundRobinLoadBalancerBuild/50000/0/1                     261 ms
benchmarkRoundRobinLoadBalancerBuild/50000/50/50                  1562 ms
benchmarkRoundRobinLoadBalancerBuild/50000/100/50                  271 ms